charters/README.md

```diff
@@ -0,0 +1,18 @@
+# Charter Guide
+
+Each HCA DCP project must have a charter that specifies its scope, responsibilities, and leadership roles.
```

Comment:
Again, we're making products.

Reply:
Assume my vote is for replacement of the word project throughout all of this PR.

Reply:
@kbergin - From my perspective, DCP is the _product_ not the individual boxes. How about _component_?

Reply:
Each box can be reused in different combinations and intend to be reusable with other products in the data biosphere. With that in mind, each one is a product. Regardless, we should be applying product thinking to the development of all of them.

Reply:
I agree with @kbergin here. I think if it was a monolithic system I would go with component but given we are making a collection of reusable software functionality where each should stand on its own (biosphere style) each piece should be able to have a life of its own and should be considered a product. The DCP in my mind is a collection of integrated products, each serving a different need in a reusable way (or so we aspire ;-) ).

Reply:
In that excerpt nothing is defined as a product. I think whether something is a component of a data biosphere environment and whether it in itself is a product are two different things, and one could be both. 

We build products that are able to be components in a data biosphere environment.

Reply:
@kbergin - That's correct. The conversation moved on. Tim responded to my suggestion to use _component_ with _given we are making a collection of reusable software functionality where each should stand on its own **(biosphere style)**_. Biosphere style is _modular components_ like the Data Store. 

In your model, is Apache Spark a product?

Reply:
@brianraymor I would like to roll back the conversation here. I indicated at the beginning of my comment that I agreed with Kylee; I do not agree the use of component is optimal as I currently understand charters. I believe, for the majority of things in charters, they will be best viewed as products (or at least existing work we have). We have so far thought of them this way so I am not sure of why that would change.

I think the reason the biopshere blog uses the word component while I think we would use product is really a question of implementation. In the biosphere excerpt, it mentions the components are served by multiple implementations. I see components here being described as the abstraction of the functionality (the class) while the products are the instantiations or what we build (the object). In reality, the biosphere will abstractly contain the functionality data store (component) that will be fulfilled/instantiated with work on many products (HCA blue box) that share similar interfaces/functionality. Given we are making the implementations I would prefer us to be product focused.

I could imagine a world where component (not biopshere component but component as in generically "a piece") could be the subject of some charters, if it is a focused augmentation of a product we have. In a way a supplemental piece. One example of this is if one wanted to charter the addition of another cloud store backend to the HCA data store. This would not necessarily be a freestanding product (like the HCA data store) but could be described as a component (if charters are that specific, still a little hazy on the scope). I hope that better communicates my thoughts.

Reply:
+1 to using reference documents from our project's history. I want a world where we can do the same with governance.

Reply:
I've done some looking around and have two thoughts here.
1. It appears that open source foundations [generally](https://www.freebsdfoundation.org/what-we-do/projects/) [use](https://www.eclipse.org/org/foundation/) [project](https://www.apache.org/foundation/how-it-works.html), although a few (with seemingly more user-facing software) use [product](https://www.mozilla.org/en-US/about/).
2. The best differentiation I've read thus far is that an `OSS product = (a software project) + standardization + QA + packaging + branding + support`.

The question to me is if each of the component teams provide the extra elements listed in (2).

Reply:
I think that much of the language we are defining has already been defined in industry. I agree that each module/box is a product in itself. Products are long-running and usually have lots of checkpoints along the way.  I would normally call these checkpoints releases, but the concept of release seems to be reserved for data releases only.  I think we can view these checkpoints/releases as projects. Each charter can be viewed as a high-level description of what will be in the release. So from my software development perspective I think that we have:
**Products:** Equivalent to our  current boxes, do not require a charter
**Projects:** Equivalent to the effort to create a software release on a product
**Charter:** High level descriptions of the organization constructed to deliver a set of features or capabilities. I think that we would like to have charters for our Product groups too, but putting charters at this granularity keeps us focused on what the Product needs to deliver feature-wise.
**Project Vision:** High level vision statements for what this release is trying to achieve. (avoid using milestone as that has the connotation of a more specific short term goal in the development jargon I am used to).

Reply:
From our [HCA Data Coordination Platform wiki](https://github.com/HumanCellAtlas/wiki/wiki):

_As currently conceived, this data coordination platform will provide four key **components**: ingestion services for submission of data; synchronized data storage across multiple clouds; standardized secondary analysis pipelines; and portals for data access, tertiary analysis, and visualization._

Reply:
I like @mweiden 's definition here that a product is the software component + a bunch of other things.  When I think about the HCA data storage service I think the software component.  When I think of Blue Box (or we call it Redwood in other projects) it's the software component + deployment scripts + documentation + slack channel/community + etc... all of which are the Blue Box (Redwood) product which is used by a wide variety of projects like HCA/Data Commons/Data STAGE/AnVIL/etc.

Anyway, this is how I'm thinking about it.  I'm pretty flexible with calling things components or products personally.  But defining it somewhere would be helpful and then staying consistent.

Reply:
I have been having some discussions and did some research and I would like to re-pitch my definition of project and product. There *are* a few key differences between commercial software and open source terminology. Virtually all open source software is referred to as projects. An open source project is often supplemented by a commercial company to make a product. Our components cannot stand alone as products. They each depend on other supporting software to become products. We depend on each other to become the DCP product.

Reply:
I really like @mweiden's definition here - this split makes total sense to me. I think our role here is to deliver the DCP as a single product. I agree for sure that there will be requirements to repurpose components - or even smaller parts of the codebase - for use in other scenarios (for example, JSON schema validation with added ontology plugins is a part of ingest we are already planning to reuse) but this will come with additional customisation or packaging to make it a standalone product.